src/Simulation/Native/src/simulator/simulator.hpp

```diff
@@ -302,6 +302,18 @@ class Simulator : public Microsoft::Quantum::Simulator::SimulatorInterface
         }
     }
 
+    void dump(TDumpToLocationCallback callback, TDumpLocation location) override
+    {
+        recursive_lock_type l(mutex());
```

Comment:
(For myself: If making changes to this PR, remove this line because of the second part of #603)